### PR TITLE
[Tizen/Api] add common utilities @open sesame 06/12 19:35

### PR DESCRIPTION
**Pull request by jaeyun-jung**
1. Define new functions to handle tensor data and refactor api functions.
2. Update api description.
3. Add testcases using custom filter.

Signed-off-by: Jaeyun Jung <jy1210.jung@samsung.com>

### CODE REVIEW

**Comment by jaeyun-jung** [comment]
PR updated (add testcases with passthrough custom filter and fix bugs)

File: tests/tizen_capi/unittest_tizen_capi.cpp [hunk 1]
```diff
@@ -1067,27 +1067,42 @@ TEST (nnstreamer_capi_singleshot, invoke_01)
...
+
   output1 = ml_model_inference (model, input, NULL);
   EXPECT_TRUE (output1 != NULL);
-  ml_model_free_tensors_data (output1);
 
-  output2 = ml_model_allocate_tensors_data (&out_info);
+  status = ml_util_get_last_error ();
+  EXPECT_EQ (status, ML_ERROR_NONE);
+
+  ml_util_free_tensors_data (&output1);
```
Comment (again4you) [inline tests/tizen_capi/unittest_tizen_capi.cpp]: `output1` is freed by `ml_util_free_tensors_data()`, then `output1` is compared with `output1` at 1092 line.

Is it correct?

File: tests/tizen_capi/unittest_tizen_capi.cpp [hunk 2]
```diff
@@ -1122,27 +1140,147 @@ TEST (nnstreamer_capi_singleshot, invoke_02)
...
```
Comment (again4you) [inline tests/tizen_capi/unittest_tizen_capi.cpp]: ditto.

File: tests/tizen_capi/unittest_tizen_capi.cpp (hunk 1 above)
Comment (again4you) [reply tests/tizen_capi/unittest_tizen_capi.cpp]: Sorry. I misunderstood. Good job!

**Comment by again4you** [comment]
LGTM!

**Comment by jaeyun-jung** [comment]
Changed name prefix for single-shot api (ml_model > ml_single)

**Pull request merged**